synapse/storage/data_stores/__init__.py

```diff
@@ -66,9 +66,9 @@ def __init__(self, main_store_class, hs):
 
                     self.main = main_store_class(database, db_conn, hs)
 
-                    # If we're on a process that can persist events (currently
-                    # master), also instantiate a `PersistEventsStore`
-                    if hs.config.worker.worker_app is None:
+                    # If we're on a process that can persist events also
+                    # instansiate a `PersistEventsStore`
+                    if hs.config.worker.writers.events == hs.get_instance_name():
```

Comment:
Ok, so `hs.get_instance_name()` is essentially how we check if we're on a "master" process, correct?

Reply:
Yeah, so each instance has a unique name and we use that to determine which process is in charge of event persistence (by default that is `"master"` but can be configured to be another instance via `writers` config)

Reply:
Makes sense, thanks!